File: content/snippets/cpp/language_basics_environment_variables.cpp
#include <cstdlib>
#include <iostream>
#include <string>

int main() {
    const char* mode_raw = std::getenv("APP_MODE");
    const char* port_raw = std::getenv("APP_PORT");

    std::string mode = mode_raw ? mode_raw : "development";
    std::string port = port_raw ? port_raw : "8080";

    std::cout << "mode: " << mode << "\n";
    std::cout << "port: " << port << "\n";
    std::cout << "has APP_MODE: " << (mode_raw != nullptr) << "\n";
    return 0;
}

File: content/snippets/scala2/language_basics_environment_variables.scala
object EnvironmentVariablesBasics {
  def main(args: Array[String]): Unit = {
    val env = sys.env
    val mode = env.getOrElse("APP_MODE", "development")
    val port = env.getOrElse("APP_PORT", "8080")

    println(s"mode: $mode")
    println(s"port: $port")
    println(s"has APP_MODE: ${env.contains("APP_MODE")}")
  }
}
